### ISSUE TITLE
⚙️  Quasar scroll assist for cordova?/capacitor!

### ISSUE THREAD

**Issue by bazuka5801** [quasarframework/quasar#9687]
**Describe the bug**
When we use capacitor in quasar, and select any input, keyboard showing, but selected input field scrolling down and we can't see what we input. We get keyboard show bug. This code snipper - fixing that bug.

Actual state of bug with quasar v1 and v2: [YouTube](https://youtu.be/3tQJakYez1Q)

**Expected behavior**
[YouTube](https://youtu.be/HEXWEVXjen8)

I write quick fix for this bug: [Gist](https://gist.github.com/bazuka5801/79acde3cb17001b76cd33d4f33ce8f6e)

But someone talk to me (and my head forgot, year ago I was testing ionic app and get confused this error is fixed), this issue already fixed in ionic.

I found this fix [Scroll Assist](https://github.com/ionic-team/ionic-framework/blob/8e0e5da7407adecb7471b3a6b0ac059337761355/core/src/utils/input-shims/hacks/scroll-assist.ts)

💪  And I want to implement it for quasar v1, v2 to fix issue with scroll! 

**Platform (please complete the following information):**
Quasar Version: 1 & 2
@quasar/app Version: 2 & 3
Quasar mode:
  - [x] Cordova
  - [x] Capacitor

Tested on:
  - [ ] Cordova
  - [x] Capacitor v3

OS: Mac OS
Yarn: Latest
Browsers: WebKit & Chrome
iOS: 12+
Android: ?

**Comment by bazuka5801** [comment]
I record a video with ionic assist to compare behavior [YouTube](https://youtu.be/rKMbGPIuKCo)

**Comment by pdanpdan** [comment]
Hello, can you please test if this fixes the problem for you?

In package.json replace quasar dependency with:
"quasar": "https://github.com/pdanpdan/quasar#quasar-pdan-v1.15.20-beta.2"

Then do a yarn or npm install and start the app again.

Please report if you find something not working.

**Comment by bazuka5801** [comment]
@pdanpdan Yes, same problem, no scrolling on webkit (capacitor 3 & ios 14.5)

I go deeper and found **scroll smooth** instead of **instant**.
I installed [smoothscroll polyfill](https://github.com/iamdustan/smoothscroll] (for webkit, chrome on android support it)
And I get this result on ios 14.5:
https://youtu.be/wDcoWqrnltY

Look nice than Ionic video.

**Comment by bazuka5801** [comment]
I made some research for capacitor keyboard plugin and found a good solution (in linked video)

[![Capacitor Keyboard - smooth open/close](http://img.youtube.com/vi/yiCnp64bkZw/0.jpg)](http://www.youtube.com/watch?v=yiCnp64bkZw "Capacitor Keyboard - Smooth open/close")

**Comment by bazuka5801** [comment]
🥳 🥳 🥳  I published an extension that fixes this issue.
https://github.com/BazuMax/capacitor-smooth-keyboard-popup

**Comment by rstoenescu** [comment]
Retested on current iOS (iPhone 17 Pro simulator, iOS 26.5, Capacitor 8) and this is no longer reproducible.

On a scrollable page (fixed header + footer, a column of outlined inputs), tapping a field near the bottom keeps it visible above the keyboard:

- Without `@capacitor/keyboard`: WebKit scrolls the document itself, so the focused field and caret stay above the keyboard.
- With `@capacitor/keyboard` (default `resize: native`): the web view shrinks and the field is again fully visible above the footer.

WebKit now reveals the focused element by itself for page-level inputs, which it did not reliably do back on iOS 12-14 when this was filed. Quasar carries no page-level keyboard-scroll code and, based on this, needs none. The dedicated keyboard handling we do have is for `QDialog` only (the dialog pins `document.body`, so the page cannot scroll to reveal the field), which is a separate case from this report.

Given that, an Ionic-style scroll-assist would only fight behavior the platform already handles, so we are not going to add one. Closing as not planned. If anyone still sees a field hidden under the keyboard on a current iOS + Capacitor/Cordova version, please reopen with a repro and the exact versions.
